lock-api/src/main/java/com/palantir/lock/client/LockLeaseService.java

```diff
@@ -47,19 +46,21 @@
 class LockLeaseService {
     private final NamespacedConjureTimelockService delegate;
     private final UUID clientId;
-    private final CoalescingSupplier<LeaderTime> time;
+    private final LeaderTimeGetter leaderTimeGetter;
     private final BlockEnforcingLockService lockService;
 
     @VisibleForTesting
-    LockLeaseService(NamespacedConjureTimelockService delegate, UUID clientId) {
+    LockLeaseService(
+            NamespacedConjureTimelockService delegate, UUID clientId, Optional<LeaderTimeGetter> leaderTimeGetter) {
         this.delegate = delegate;
         this.clientId = clientId;
-        this.time = new CoalescingSupplier<>(delegate::leaderTime);
+        this.leaderTimeGetter = leaderTimeGetter.orElseGet(() -> new CoalescingLeaderTimeGetter(delegate));
         this.lockService = BlockEnforcingLockService.create(delegate);
     }
 
-    static LockLeaseService create(NamespacedConjureTimelockService conjureTimelock) {
-        return new LockLeaseService(conjureTimelock, UUID.randomUUID());
+    static LockLeaseService create(
+            NamespacedConjureTimelockService conjureTimelock, Optional<LeaderTimeGetter> leaderTimeGetter) {
```

Comment:
I would make this non-Optional actually: the setting of a default should happen at the TransactionManagers builder level, and then all usages within Atlas code should be non-optional (e.g. see how `lockWatchingCache` is set)